packages/@o3r/rules-engine/src/engine/ruleset-executor.ts

```diff
@@ -61,6 +62,137 @@ export class RulesetExecutor {
     this.engineRuleset = this.prepareRuleset();
   }
 
+
+  /**
+   * Plug ruleset to fact streams and trigger a first evaluation
+   */
+  private prepareRuleset() {
+    this.ruleset.rules.forEach((rule) => {
+      const findRuleInputFacts = (obj: TopLevelCondition): string[] => {
+        const ruleInputFacts: string[] = [];
+        // Recursive function to explore each object
+        const exploreObject = (currentObj: any) => {
+          if (currentObj.type === 'FACT') {
+            ruleInputFacts.push(currentObj.value);
+          } else {
+            for (const key in currentObj) {
+              if (key === 'operator') {
+                const op = this.operators[currentObj[key]];
+                if (op && op.dependencies) {
+                  ruleInputFacts.push(...op.dependencies);
+                }
+              } else if (typeof currentObj[key] === 'object' && currentObj[key] !== null) {
+                exploreObject(currentObj[key]);
+              }
+            }
+          }
+        };
+        exploreObject(obj);
+        return Array.from(new Set(ruleInputFacts));
+      };
+      rule.inputFacts = findRuleInputFacts(rule.rootElement.condition);
```

Comment:
`rule` can be read-only in some cases (when coming from the store with strictImmutability)
you may need to store it somewhere else